diff -rupN --no-dereference postgresql-15.1/src/interfaces/ecpg/ecpglib/misc.c postgresql-15.1-new/src/interfaces/ecpg/ecpglib/misc.c
--- postgresql-15.1/src/interfaces/ecpg/ecpglib/misc.c	2022-11-07 22:36:53.000000000 +0100
+++ postgresql-15.1-new/src/interfaces/ecpg/ecpglib/misc.c	2023-01-07 15:20:57.358278879 +0100
@@ -450,7 +450,7 @@ ECPGis_noind_null(enum ECPGttype type, c
 	return false;
 }
 
-#ifdef WIN32
+#ifdef _MSC_VER
 #ifdef ENABLE_THREAD_SAFETY
 
 void
diff -rupN --no-dereference postgresql-15.1/src/interfaces/ecpg/include/ecpg-pthread-win32.h postgresql-15.1-new/src/interfaces/ecpg/include/ecpg-pthread-win32.h
--- postgresql-15.1/src/interfaces/ecpg/include/ecpg-pthread-win32.h	2022-11-07 22:36:53.000000000 +0100
+++ postgresql-15.1-new/src/interfaces/ecpg/include/ecpg-pthread-win32.h	2023-01-07 15:20:57.358278879 +0100
@@ -7,7 +7,7 @@
 
 #ifdef ENABLE_THREAD_SAFETY
 
-#ifndef WIN32
+#ifndef _MSC_VER
 
 #include <pthread.h>
 #else
diff -rupN --no-dereference postgresql-15.1/src/interfaces/libpq/fe-connect.c postgresql-15.1-new/src/interfaces/libpq/fe-connect.c
--- postgresql-15.1/src/interfaces/libpq/fe-connect.c	2022-11-07 22:36:53.000000000 +0100
+++ postgresql-15.1-new/src/interfaces/libpq/fe-connect.c	2023-01-07 15:20:57.359278887 +0100
@@ -56,7 +56,7 @@
 #endif
 
 #ifdef ENABLE_THREAD_SAFETY
-#ifdef WIN32
+#ifdef _MSC_VER
 #include "pthread-win32.h"
 #else
 #include <pthread.h>
diff -rupN --no-dereference postgresql-15.1/src/interfaces/libpq/fe-secure.c postgresql-15.1-new/src/interfaces/libpq/fe-secure.c
--- postgresql-15.1/src/interfaces/libpq/fe-secure.c	2022-11-07 22:36:53.000000000 +0100
+++ postgresql-15.1-new/src/interfaces/libpq/fe-secure.c	2023-01-07 15:20:57.360278895 +0100
@@ -38,7 +38,7 @@
 #include <sys/stat.h>
 
 #ifdef ENABLE_THREAD_SAFETY
-#ifdef WIN32
+#ifdef _MSC_VER
 #include "pthread-win32.h"
 #else
 #include <pthread.h>
diff -rupN --no-dereference postgresql-15.1/src/interfaces/libpq/fe-secure-openssl.c postgresql-15.1-new/src/interfaces/libpq/fe-secure-openssl.c
--- postgresql-15.1/src/interfaces/libpq/fe-secure-openssl.c	2022-11-07 22:36:53.000000000 +0100
+++ postgresql-15.1-new/src/interfaces/libpq/fe-secure-openssl.c	2023-01-07 15:20:57.360278895 +0100
@@ -47,7 +47,7 @@
 #include <sys/stat.h>
 
 #ifdef ENABLE_THREAD_SAFETY
-#ifdef WIN32
+#ifdef _MSC_VER
 #include "pthread-win32.h"
 #else
 #include <pthread.h>
diff -rupN --no-dereference postgresql-15.1/src/interfaces/libpq/libpq-int.h postgresql-15.1-new/src/interfaces/libpq/libpq-int.h
--- postgresql-15.1/src/interfaces/libpq/libpq-int.h	2022-11-07 22:36:53.000000000 +0100
+++ postgresql-15.1-new/src/interfaces/libpq/libpq-int.h	2023-01-07 15:20:57.360278895 +0100
@@ -29,7 +29,7 @@
 #endif
 
 #ifdef ENABLE_THREAD_SAFETY
-#ifdef WIN32
+#ifdef _MSC_VER
 #include "pthread-win32.h"
 #else
 #include <pthread.h>
diff -rupN --no-dereference postgresql-15.1/src/interfaces/libpq/Makefile postgresql-15.1-new/src/interfaces/libpq/Makefile
--- postgresql-15.1/src/interfaces/libpq/Makefile	2022-11-07 22:36:53.000000000 +0100
+++ postgresql-15.1-new/src/interfaces/libpq/Makefile	2023-01-07 15:20:57.361278903 +0100
@@ -71,9 +71,6 @@ ifeq ($(PORTNAME), win32)
 OBJS += \
 	win32.o
 
-ifeq ($(enable_thread_safety), yes)
-OBJS += pthread-win32.o
-endif
 endif
 
 
