diff -rupN --no-dereference postgresql-14.3/src/pl/tcl/Makefile postgresql-14.3-new/src/pl/tcl/Makefile
--- postgresql-14.3/src/pl/tcl/Makefile	2022-05-09 23:14:45.000000000 +0200
+++ postgresql-14.3-new/src/pl/tcl/Makefile	2022-11-18 09:51:05.235025440 +0100
@@ -16,6 +16,10 @@ override CPPFLAGS := -I. -I$(srcdir) $(T
 # On Windows, we don't link directly with the Tcl library; see below
 ifneq ($(PORTNAME), win32)
 SHLIB_LINK = $(TCL_LIB_SPEC) $(TCL_LIBS) -lc
+else
+ifeq ($(host_os), mingw32)
+SHLIB_LINK = $(TCL_LIB_SPEC) $(TCL_LIBS)
+endif
 endif
 
 PGFILEDESC = "PL/Tcl - procedural language"
@@ -36,6 +40,7 @@ REGRESS = pltcl_setup pltcl_queries pltc
 # which are not compatible with mingw gcc. Therefore we need to build a
 # new import library to link with.
 ifeq ($(PORTNAME), win32)
+ifneq ($(host_os), mingw32)
 
 tclwithver = $(subst -l,,$(filter -l%, $(TCL_LIB_SPEC)))
 TCLDLL = $(dir $(TCLSH))/$(tclwithver).dll
@@ -48,6 +53,7 @@ lib$(tclwithver).a: $(tclwithver).def
 $(tclwithver).def: $(TCLDLL)
 	gendef - $^ > $@
 
+endif # mingw32
 endif # win32
 
 
@@ -96,8 +102,10 @@ clean distclean: clean-lib
 	rm -f $(OBJS)
 	rm -rf $(pg_regress_clean_files)
 ifeq ($(PORTNAME), win32)
+ifneq ($(host_os), mingw32)
 	rm -f $(tclwithver).def
 endif
+endif
 
 maintainer-clean: distclean
 	rm -f pltclerrcodes.h
